pgmpy/models/DoubleMLRegressor.py

```diff
@@ -0,0 +1,188 @@
+            g_hat = np.repeat(y_vec.mean(), len(y_vec))
+            self.estimator_g_ = None
+        else:
+            ml_g.fit(X_for_g, y_vec)
+            self.estimator_g_ = ml_g
+            g_hat = ml_g.predict(X_for_g)
+            g_hat = np.asarray(g_hat).ravel()
+
+        # fit m (treatment mechanism)
+        t_vec = df[treat_col].to_numpy(dtype=float)
+        if X_for_g.shape[1] == 0:
+            m_hat = np.repeat(t_vec.mean(), len(t_vec))
+            self.estimator_m_ = None
+        else:
+            ml_m.fit(X_for_g, t_vec)
+            self.estimator_m_ = ml_m
+            # if classifier, prefer predict_proba for propensity
+            if hasattr(ml_m, "predict_proba"):
+                try:
+                    probs = ml_m.predict_proba(X_for_g)
+                    # if binary, take second column
+                    if probs.ndim == 2 and probs.shape[1] >= 2:
+                        m_hat = probs[:, 1]
+                    else:
+                        m_hat = probs.ravel()
+                except Exception:
+                    m_hat = ml_m.predict(X_for_g)
+            else:
+                m_hat = ml_m.predict(X_for_g)
+            m_hat = np.asarray(m_hat).ravel()
+
+        self.g_hat_ = g_hat
+        self.m_hat_ = m_hat
+
+        # orthogonalization
+        y_res = y_vec - self.g_hat_
+        t_res = t_vec - self.m_hat_
+
+        # regress y_res on t_res (with intercept)
+        X_res = np.column_stack([np.ones(len(t_res)), t_res])
+        theta_coef, *_ = np.linalg.lstsq(X_res, y_res, rcond=None)
+
+        intercept = float(theta_coef[0])
+        theta = float(theta_coef[1])  # single-treatment effect
+
+        # store both a clear treatment_effect_ and a coef_ vector
+        self.treatment_effect_ = theta
+        # create coef_ vector aligned with design_cols
+        # position 0 is treatment, rest correspond to adjustments
+        self.coef_ = np.concatenate(
+            ([theta], np.zeros(len(design_cols) - 1, dtype=float))
+        )
+        self.intercept_ = intercept
+
+        self._design_columns = design_cols.copy()
+        self.is_fitted_ = True
+        return self
+
+    def predict(self, X):
+        """Return predictions from the fitted internal linear model."""
+        check_is_fitted(self, "is_fitted_")
+        X_df = self._as_dataframe(X)
+
+        missing = [c for c in self._design_columns if c not in X_df.columns]
+        if missing:
+            n_needed = len(self._design_columns)
+            if X_df.shape[1] < n_needed:
+                raise ValueError(
+                    f"Predict input has fewer columns than required: "
+                    f"need {n_needed}, got {X_df.shape[1]}"
+                )
+            mapped_cols = list(X_df.columns[:n_needed])
+        else:
+            mapped_cols = self._design_columns
+
+        X_design = X_df[mapped_cols].to_numpy(dtype=float)
+        preds = self.intercept_ + X_design.dot(self.coef_)
```

Comment:
This should be: `preds = self.intercept_ + self.treatment_effect_ * X[exposure_variable] + ml_g.predict(X[adjustment])`